hydra-node/src/Hydra/API/Server.hs

```diff
@@ -89,18 +90,27 @@ withAPIServer ::
 withAPIServer config env party persistence tracer chain pparams serverOutputFilter callback action =
   handle onIOException $ do
     responseChannel <- newBroadcastTChanIO
-    timedOutputEvents <- loadAll
-
-    -- Intialize our read model from stored events
-    headStatusP <- mkProjection Idle (output <$> timedOutputEvents) projectHeadStatus
-    snapshotUtxoP <- mkProjection Nothing (output <$> timedOutputEvents) projectSnapshotUtxo
-    commitInfoP <- mkProjection CannotCommit (output <$> timedOutputEvents) projectCommitInfo
-    headIdP <- mkProjection Nothing (output <$> timedOutputEvents) projectInitializingHeadId
-    pendingDepositsP <- mkProjection [] (output <$> timedOutputEvents) projectPendingDeposits
-
+    -- Intialize our read models from stored events
+    -- NOTE: we do not keep the stored events around in memory
+    headStatusP <- mkProjection Idle projectHeadStatus
+    snapshotUtxoP <- mkProjection Nothing projectSnapshotUtxo
+    commitInfoP <- mkProjection CannotCommit projectCommitInfo
+    headIdP <- mkProjection Nothing projectInitializingHeadId
+    pendingDepositsP <- mkProjection [] projectPendingDeposits
+    _ <-
+      runConduitRes $
+        source
+          .| mapC output
+          .| mapM_C (lift . atomically . update headStatusP)
+          .| mapM_C (lift . atomically . update snapshotUtxoP)
+          .| mapM_C (lift . atomically . update commitInfoP)
+          .| mapM_C (lift . atomically . update headIdP)
+          .| mapM_C (lift . atomically . update pendingDepositsP)
```

Comment:
@v0d1ch `mapM_C` is consuming events.. so not all functions would see the same outputs! See https://hackage.haskell.org/package/conduit-1.3.6/docs/Conduit.html#v:mapM_C

Reply:
You are right, I used `iterM` instead and just did `mapM_C` in the final pipeline. One of the tests caught this and now with your suggestion it works again.

Reply:
as a bonus I added streaming to history TVar to resolve another fixme